lib/redis.rb

```diff
@@ -2852,46 +2848,49 @@ def method_missing(command, *args)
   # returning false.
   Boolify =
     lambda { |value|
-      value == 1 if value
+      case value
+      when 1 then true
+      when 0 then false
+      else value
+      end
     }
 
   BoolifySet =
     lambda { |value|
-      if value && "OK" == value
-        true
-      else
-        false
+      case value
+      when "OK" then true
+      when nil then false
+      else value
       end
     }
 
   Hashify =
-    lambda { |array|
+    lambda { |value|
+      return value unless value.respond_to?(:each_slice)
+
       hash = Hash.new
-      array.each_slice(2) do |field, value|
-        hash[field] = value
+      value.each_slice(2) do |key, val|
+        hash[key] = val
       end
       hash
     }
 
   Floatify =
-    lambda { |str|
-      if str
-        if (inf = str.match(/^(-)?inf/i))
-          (inf[1] ? -1.0 : 1.0) / 0.0
-        else
-          Float(str)
-        end
+    lambda { |value|
+      case value
+      when "inf" then 1.0 / 0.0
+      when "-inf" then -1.0 / 0.0
```

Comment:
Maybe we could use `Float::INFINITY` instead?

Reply:
Thanks! Updated